packages/dashboard-core-plugins/src/panels/CorePanel.tsx

```diff
@@ -0,0 +1,80 @@
+import React, { PureComponent, type ReactElement } from 'react';
+import { createXComponent } from '@deephaven/components';
+import { type BasePanelProps, BasePanel } from '@deephaven/dashboard';
+import type { dh } from '@deephaven/jsapi-types';
+import { ConsoleEvent, InputFilterEvent } from '../events';
+
+export type CorePanelProps = BasePanelProps & {
+  onClearAllFilters?: (...args: unknown[]) => void;
+  onSessionClose?: (session: dh.IdeSession) => void;
+  onSessionOpen?: (
+    session: dh.IdeSession,
+    { language, sessionId }: { language: string; sessionId: string }
+  ) => void;
+};
+
+/**
+ * Generic panel component that emits mount/unmount/focus events.
+ * Also wires up some triggers for common events:
+ * Focus, Resize, Show, Session open/close, client disconnect/reconnect.
+ */
+class CorePanel extends PureComponent<CorePanelProps> {
```

Comment:
~~Doesn't handle focus, resize, show, or client disconnect/reconnect? Update comment or add those events~~ Nevermind I see this renders `BasePanel` inside of it

Also, just to clarify. "Core" Panel is talking about DH Core?
`CorePanel` is in `dashboard-core-plugins`. Could call `DHCPanel` or just `DHPanel`, but I'm indifferent
`Panel` is in `Dashboard` which seems to handle the other events mentioned in this comment